Analysis/src/Frontend.cpp

```diff
@@ -1397,7 +1398,9 @@ ModulePtr Frontend::check(const SourceModule& sourceModule, Mode mode, std::vect
     }
     else
     {
-        TypeChecker typeChecker(globals.globalScope, forAutocomplete ? &moduleResolverForAutocomplete : &moduleResolver, builtinTypes, &iceHandler);
+        TypeChecker typeChecker(FFlag::LuauTypeCheckerUseCorrectScope ? (forAutocomplete ? globalsForAutocomplete.globalScope : globals.globalScope)
```

Comment:
Not sure if this should actually use `environmentScope` like the new solver does above? It seems that environment scope does correctly point to globalScopeForAutocomplete in this case

Reply:
It shouldn't, `environmentScope` in the old solver is handled in the `check` function.